src/ipa_pc/data_structures.rs

```diff
@@ -263,29 +383,85 @@ pub struct BatchProof<G: AffineCurve> {
     pub batch_values: BTreeMap<String, G::ScalarField>
 }
 
-impl<G: AffineCurve> BatchPCProof for BatchProof<G> {
-    fn size_in_bytes(&self) -> usize {
-        to_bytes![self].unwrap().len()
+impl<G: AffineCurve> SemanticallyValid for BatchProof<G> {
+    fn is_valid(&self) -> bool {
+        self.proof.is_valid() &&
+            self.batch_commitment.is_valid() &&
+            self.batch_values.iter().all(|(_, v)| v.is_valid())
```

Comment:
Here, we could check if the number of ``batch_values`` correspond to the (overall) number of commitments (including the ``batch_commitment``)

Reply:
I don't have access to the commitments here